src/freenet/client/async/ClientPutter.java

```diff
@@ -287,7 +287,7 @@ public void onSuccess(ClientPutState state, ClientContext context) {
 		}
 		if(super.failedBlocks > 0 || super.fatallyFailedBlocks > 0 || super.successfulBlocks < super.totalBlocks) {
 			// USK auxiliary inserts are allowed to fail.
-			if(!uri.isUSK())
+			if((!uri.isUSK()) && !ctx.getCHKOnly)
```

Comment:
The auxiliary blocks are not inserted at all, we just generate the keys for the blocks and then stop, so the block counts might be inconsistent. For a normal insert we'd generate the keys *then* insert them, and report them on the actual insert. Would it be better for SplitFileInserter to lie about the counts in this case?